### PR TITLE
macfuse-stubs: replace osxfuse 3.8.3

### PR DESCRIPTION
**Pull request by midchildan**
###### Motivation for this change
macFUSE is a rebranded version of osxfuse, and is the same project.

This PR, on the other hand, greatly changes how FUSE applications are built for macOS. macFUSE (and also osxfuse) mainly consists of two parts, a kernel extension and the libfuse userspace library. Since macOS kernel extensions aren't supported on Nix, we only build libfuse and expect users to install macFUSE outside of Nix to obtain the kernel extension. However, this approach is problematic for a couple of reasons:

1. libfuse and the rest of macFUSE work closely toghether, so there would be problems when the versions don't match
2. macFUSE's libfuse repostitory doesn't have tags for recent releases
3. The libfuse source is public in part for LGPL compliance, and I'm not sure if there would be a continuing effort to make sure it's easily buildable

While points 2 and 3 might be fixable by contacting upstream, there's not much that could be done about 1 with the current approach. To fix this problem, this PR drops our own libfuse build and replaces it with a build-time stub library generated from the official macFUSE release binaries. At runtime, applications would load the libfuse binary included with official macFUSE. Users are still expected to install macFUSE itself from the official source. This is the same approach currently taken for macOS system libraries with the `darwin-stubs` package.

It should be noted that parts of macFUSE including the kernel extension are distributed under a license that restricts commercial redistribution. However, the proposed package doesn't touch those parts and only concerns itself with libfuse, [which is LGPL](https://github.com/osxfuse/osxfuse/blob/macfuse-4.1.0/LICENSE.txt#L212-L215). This should allow Nix to continue to provide binary caches for FUSE-dependant packages on macOS.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19312,7 +19312,9 @@ in
     inherit (pkgs.darwin.apple_sdk.frameworks) IOKit;
   };
 
-  osxfuse = callPackage ../os-specific/darwin/osxfuse { };
+  macfuse-stubs = callPackage ../os-specific/darwin/macfuse {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:19315]: Please add an alias to aliases.nix.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Also please take a look at the eval error.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (midchildan) [reply pkgs/top-level/all-packages.nix:19377]: Fixed.

**Comment by midchildan** [comment]
It looks like I missed a few packages that depended on osxfuse. I fixed those and checked that these additional packages works with macfuse.

**Comment by lovesegfault** [comment]
Could you add yourself as a maintainer of `macfuse`?

**Comment by midchildan** [comment]
Oops. Fixed.

**Pull request merged**